Mathlib/Data/Real/EReal.lean

```diff
@@ -832,6 +865,20 @@ theorem add_lt_top {x y : EReal} (hx : x ≠ ⊤) (hy : y ≠ ⊤) : x + y < ⊤
   rw [← EReal.top_add_top]
   exact EReal.add_lt_add hx.lt_top hy.lt_top
 
+lemma add_ne_top {x y : EReal} (hx : x ≠ ⊤) (hy : y ≠ ⊤) : x + y ≠ ⊤ := by
+  rw [← lt_top_iff_ne_top]
+  exact add_lt_top hx hy
+
+lemma add_ne_top_iff_of_ne_bot {x y : EReal} (hx : x ≠ ⊥) (hy : y ≠ ⊥) :
+    x + y ≠ ⊤ ↔ x ≠ ⊤ ∧ y ≠ ⊤ := by
+  refine ⟨?_, fun h ↦ add_ne_top h.1 h.2⟩
+  induction x <;> simp_all
+  induction y <;> simp_all
+
+lemma add_ne_top_iff_of_ne_bot_of_ne_top {x y : EReal} (hy : y ≠ ⊥) (hy' : y ≠ ⊤) :
```

Comment:
I don't really understand the name of this lemma

Reply:
Same as above. 
Here maybe we could name it something like `add_ne_top_iff_of_finite` or `add_finite_ne_top_iff_ne_top`, since we are basically asking for `y` to be finite, but I'm not sure these names are actually acceptable, because nothing named `finite` actually appears in the statement...

Reply:
I agree that `finite` is not an option, but perhaps my suggestion above helps?

Reply:
Option 2. (with the addition) is certainly the best!